docs/project/glossary.md

```diff
@@ -232,34 +127,58 @@ officially updated since .NET Framework 2.0.
 
 ### Mono
 
-Mono is an open source alternative to the .NET Framework. Mono started around
+[Mono][mono] is an open source alternative to the .NET Framework. Mono started around
 the same time the .NET Framework was first released. Since Microsoft didn't
-release Rotor as open source, Mono was forced to start from scratch and is thus
+release Rotor as open source, Mono was forced to start from scratch and was thus
 a complete re-implementation of the .NET Framework with no shared code.
 
-When .NET Core was released under the MIT license, Microsoft also released large
-chunks of the .NET Framework under the MIT license as well, which can be found
-[here][referencesource]. This enabled the Mono community to use the same code
-the .NET Framework uses in order to close gaps and avoid behavioral differences.
+Today, the [Mono runtime](https://github.com/dotnet/runtime/tree/master/src/mono) is part
+of the unified .NET platform. It is optimized for mobile (e.g. Xamarin) and browser (e.g. Blazor) scenarios.
+
+### CoreCLR
+
+Originally, CoreCLR was the runtime of Silverlight and was designed to run on multiple
+platforms, specifically Windows and OS X.
+
+Today, the [CoreCLR runtime](https://github.com/dotnet/runtime/tree/master/src/coreclr) 
+is part of unified .NET platform. It is optimized for cloud (e.g. ASP.NET) and
+desktop (e.g. WinForms, WPF) scenarios.
+
+## Ahead-Of-Time Compilation (AOT)
+
+Most flavors of .NET runtime come with at least partial AOT compilation. Variety of AOT technologies 
+with unique characteristics were developed for .NET runtimes over the years.
+
+### ReadyToRun
+
+**Also referred to as**: R2R
+
+[ReadyToRun](https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/botr/readytorun-overview.md)
+is a file format used by CoreCLR runtime to store AOT compiled code. `crossgen` is the AOT compiler that
+produces binaries in the ReadyToRun file format.
+
+### NGen
+
+[NGen](https://docs.microsoft.com/en-us/dotnet/framework/tools/ngen-exe-native-image-generator)
+is AOT technology included in .NET Framework. It usually compiles code on the machine where
+the code will be executed at install time.
+
+### Full AOT
 
-Mono is primarily used to run .NET applications on Linux and macOS (though to
-get into the Mac App Store you need Xamarin, see below). There are ports of Mono
-to other platforms, see [Mono's Supported Platforms][mono-supported-platforms]
+[Full AOT](https://docs.microsoft.com/en-us/xamarin/ios/internals/architecture) is used
+by Mono runtime in environments that prohibit fallback to JIT.
 
-Mono has implementations (though not necessarily complete) of WinForms, ASP.NET,
-and System.Drawing.
+### Hybrid AOT
 
-### Xamarin
+[Hybrid AOT](https://docs.microsoft.com/en-us/xamarin/mac/internals/aot#hybrid-aot) is used
```

Comment:
[Hybrid AOT](https://docs.microsoft.com/en-us/xamarin/mac/internals/aot#hybrid-aot) is used
by Mono runtime in the environment that allows fallback to JIT or environment which requires read-execute protection and still need to execute IL code at runtime.